index.js

```diff
@@ -20,4 +20,44 @@ if (typeof window !== 'undefined') {
       navLinks.classList.toggle('active');
     });
   });
-}
\ No newline at end of file
+}
+const contactForm = document.querySelector('.contact-form');
+contactForm.addEventListener('submit', (e) => {
+  alert('Message sent successfully');
+  e.preventDefault();
+  const name = document.querySelector('#name').value;
+  const text = document.querySelector('#text').value;
+  const email = document.querySelector('#email').value;
+  const textarea = document.querySelector('#textarea').value;
+  if (name.length === 0) {
+    document.querySelector('.error-name').innerHTML = 'Please enter a name';
+    document.querySelector('.error-name').style.color = 'red';
+  } else if (text.length === 0) {
+    document.querySelector('.error-text').innerHTML = 'Please enter your Last name';
+    document.querySelector('.error-text').style.color = 'red';
+  } else if (email.length === 0) {
+    document.querySelector('.error-email').innerHTML = 'Please enter a valid email';
+    document.querySelector('.error-email').style.color = 'red';
+  } else if (textarea.length === 0) {
+    document.getElementById('#error_textarea').innerHTML = 'Please enter a query';
+    document.querySelector('#error_textarea').style.color = 'red';
+  } else {
+    // contactForm.submit();
+    // add data to localstorage
+    const messageObject = {
+      visitorname: name,
+      visitorlastname: text,
+      visitoremail: email,
+      message: textarea,
+    };
+    if (localStorage.getItem('messages')) {
+      // append
+      const mymessages = JSON.parse(localStorage.getItem('messages')) || [];
+      mymessages.push(messageObject);
+      localStorage.setItem('messages', JSON.stringify(mymessages));
```

Comment:
- [ ] The contact form information is persistent to page reloads so kindly implement it as well. eg
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```